.github/workflows/build_docs.yml

```diff
@@ -1,21 +1,39 @@
 ---
-name: Publish docs via Github Pages
-
-on:  # yamllint disable-line rule:truthy
+name: Deploy Latest Documentation
+on:
   workflow_dispatch:
+  release:
+    types: [published]
   push:
-    branches: [main]
     paths:
       - 'docs/**'
-
+      - mkdocs.yml
+    branches:
+      - main
 jobs:
-  build:
-    name: Deploy docs
+  deploy:
+    name: Deploy Latest Documentation
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout main
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: true
       - uses: actions/setup-python@v4
         with:
           python-version: 3.x
-      - run: pip install -r docs/requirements.txt
-      - run: mkdocs gh-deploy --strict --force
+      - name: Install Dependencies
+        run: |
+          pip install -r docs/requirements.txt
+      - name: Configure Git User
+      # Required by mike for the commit it does to the gh-pages branch
+        run: |
+          git config user.name "ci-docs"
+          git config user.email "ci-docs@dummy.com"
+      - name: Deploy the latest documents from new release
+        if: ${{ github.event_name == 'release' }}
+        run: mike deploy --push --update-aliases ${GITHUB_REF#refs/*/} latest-release
+      - name: Deploy the development docs
+        if: ${{ github.event_name == 'push' }}
```

Comment:
this means the workflow_dispatch won't do anything...
maybe you should check if the event != release?